### PR TITLE
[WIP] MergeAggr when single node has multiple batches

### LINKED ISSUE #1160: [BUG] Tpch query 06 fails when MAX_DATA_LOAD_CONCAT_CACHE_BYTE_SIZE = 10
**Describe the bug**
when using `MAX_DATA_LOAD_CONCAT_CACHE_BYTE_SIZE = 10` as `config_options` param in the `BlazingContext` the result fails.

**Steps/Code to reproduce bug**

```
from blazingsql import BlazingContext

conf_opt_1 = {}
conf_opt_1['MAX_DATA_LOAD_CONCAT_CACHE_BYTE_SIZE'] = 10

bc = BlazingContext(config_options=conf_opt_1)

bc.create_table('lineitem', '/../dataset/100MB2Part/tpch/lineitem_*.parquet')

query = """
        select
            sum(l_extendedprice*l_discount) as revenue
        from
            lineitem
        where
            l_shipdate >= date '1994-01-01'
            and l_shipdate < date '1994-01-01' + interval '1' year
            and l_discount between 0.06 - 0.01 and 0.06 + 0.01
            and l_quantity < 24
    """
print(bc.sql(query))
```

---------- Results -----------
Using default `config_options` params (correct):
```
        revenue
0  1.180342e+07
```

Using `MAX_DATA_LOAD_CONCAT_CACHE_BYTE_SIZE = 10` (wrong):
```
        revenue
0  5.981990e+06
1  5.821430e+06
```

**Expected behavior**
using different values for MAX_DATA_LOAD_CONCAT_CACHE_BYTE_SIZE, the result must be the same.

**Environment overview (please complete the following information)**
 - Environment location: [Bare-metal]
 - Method of BlazingSQL install: [from source]

**Additional context**
The same issue could be related to `tpch-queries` {14, 16, 19} 
 NOTE: the sum of both (wrong values) seems to be the right value

### PR DESCRIPTION
**Pull request by Christian8491**
This PR closes #1160  and #1161

### CODE REVIEW

**Comment by Christian8491** [comment]
rerun tests

File: engine/src/execution_graph/logic_controllers/BatchAggregationProcessing.h [hunk 1]
```diff
@@ -316,7 +316,7 @@ class MergeAggregateKernel : public kernel {
                         concatenated->toBlazingTableView(), mod_group_column_indices);
             } else if (group_column_indices.size() == 0) {
                 // aggregations without groupby are only merged on the master node
-                if(context->isMasterNode(ral::communication::CommunicationData::getInstance().getSelfNode())) {
+                if( (context->getTotalNodes() == 1) ||  (context->isMasterNode(ral::communication::CommunicationData::getInstance().getSelfNode()))  ) {
```
Comment (williamBlazing) [inline engine/src/execution_graph/logic_controllers/BatchAggregationProcessing.h:319]: This looks like a band-aid. We need to instead figure out why if there is only one node, its not automatically the MasterNode

**Review by williamBlazing — CHANGES_REQUESTED** [review]

**Comment by Christian8491** [comment]
rerun tests

**Comment by ajschmidt8** [comment]
rerun tests

**Comment by Christian8491** [comment]
rerun tests

**Comment by ajschmidt8** [comment]
@Christian8491, Ops guy here. I reran these tests one more time. Please make sure your PR includes the changes from #1168 to prevent SSL errors.

**Comment by ajschmidt8** [comment]
rerun tests

**Pull request merged**